pkg/arm/data/azuredeploy.json

```diff
@@ -302,158 +240,109 @@
-                    "networkInterfaces": [
-                        {
-                            "id": "[resourceId('Microsoft.Network/networkInterfaces', concat('nic-compute-', copyIndex()))]"
-                        }
-                    ]
-                },
-                "osProfile": {
-                    "adminUsername": "cloud-user",
-                    "computername": "[concat('vm-compute-', copyIndex())]",
-                    "linuxConfiguration": {
-                        "disablePasswordAuthentication": true,
-                        "ssh": {
-                            "publicKeys": [
-                                {
-                                    "keyData": "{{ SSHPublicKeyAsString .Config.SSHKey.PublicKey }}",
-                                    "path": "/home/cloud-user/.ssh/authorized_keys"
-                                }
-                            ]
+                "virtualMachineProfile":{
+                    "osProfile": {
+                        "adminUsername": "cloud-user",
+                        "computerNamePrefix":"compute-",
+                        "linuxConfiguration": {
+                            "disablePasswordAuthentication": true,
+                            "ssh": {
+                                   "publicKeys": [
+                                        {
+                                           "keyData": "{{ SSHPublicKeyAsString .Config.SSHKey.PublicKey }}",
+                                           "path": "/home/cloud-user/.ssh/authorized_keys"
+                                       }
+                                    ]
+                            }
                         }
-                    }
-                },
-                "storageProfile": {
-                    "imageReference": {
+                    },
+                    "storageProfile": {
+                        "imageReference": {
 {{ if eq .Config.ImageResourceName "" }}
-                        "offer": "{{ .Config.ImageOffer }}",
-                        "publisher": "{{ .Config.ImagePublisher }}",
-                        "sku": "{{ .Config.ImageSKU }}",
-                        "version": "{{ .Config.ImageVersion }}"
+                            "offer": "{{ .Config.ImageOffer }}",
+                            "publisher": "{{ .Config.ImagePublisher }}",
+                            "sku": "{{ .Config.ImageSKU }}",
+                            "version": "{{ .Config.ImageVersion }}"
 {{ else }}
-                        "id": "[resourceId('{{ .Config.ImageResourceGroup }}', 'Microsoft.Compute/images', '{{ .Config.ImageResourceName }}')]"
+                            "id": "[resourceId('{{ .Config.ImageResourceGroup }}', 'Microsoft.Compute/images', '{{ .Config.ImageResourceName }}')]"
 {{ end }}
+                        },
+                        "osDisk": {
+                            "caching": "ReadWrite",
+                            "createOption": "FromImage"
```

Comment:
Looks like the scaleset is coming up with HDDs, not SSDs.  I think you might need `"managedDisk": { "storageAccountType": "Premium_LRS" }` here to fix that.
          },

Reply:
our default for all is ssd?

Reply:
Yes.  Non-configurable.  Storage is Azure's Achilles heel.